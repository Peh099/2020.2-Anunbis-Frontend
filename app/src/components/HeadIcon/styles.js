import styled, {keyframes} from 'styled-components';

const fade = keyframes`
    0% {
        transform: translateX(20%)
    }
    100% {
        transform: translateX(0%)
    }
`;



export const Picture = styled.div`
    display: flex;
    position: relative;
    width: 150px;
    height: 150px;
    overflow: hidden;
    animation: ${fade} 3s ease;

    a {
        color: inherit;
        height: 100%;
        width: 100%;
        position: absolute;
        text-decoration: none;
    }

    :hover {
        img {
            border-radius: 50px;   
        }
        div {
            opacity: 1;
        }
    }

    img {
        height: 100%;
        width: 100%;
        border-radius: 50%;
        transition: all 0.5s ease;
        border: 1px solid;
    }

    div {
        font-size: 16px;
        font-weight: bold;
        text-shadow: 2px 2px 2px black;
        color: var(--white);
        height: 100%;
        width: 100%;
        position: absolute;
        display: flex;
        justify-content: center;
        align-items: center;
        transition: all 0.5s ease-in-out;
        opacity: 0;
    }
    
`