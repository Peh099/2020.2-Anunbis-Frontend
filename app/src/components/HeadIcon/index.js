import React from 'react';
import { Picture } from './styles';


export default function HeadIcon({ imgSrc, name, linkGithub }) {

    return (
        <Picture>

            <a target="_blank" href={linkGithub} >
            <div>
                    {name}
            </div>
            <img src={imgSrc} alt={name} />

            </a>
        </Picture>
    );
}