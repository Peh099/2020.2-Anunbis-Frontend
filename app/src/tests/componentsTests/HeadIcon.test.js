import renderer from 'react-test-renderer';
import HeadIcon from '../../components/HeadIcon';

describe('Snapshot HeadIcon component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<HeadIcon />).toJSON() 
    expect(tree).toMatchSnapshot()
    })
})