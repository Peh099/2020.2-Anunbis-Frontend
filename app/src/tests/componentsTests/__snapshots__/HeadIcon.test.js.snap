// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot HeadIcon component matches the snapshot 1`] = `
<div
  className="sc-bdnxRM gvdwan"
>
  <a
    rel="noreferrer"
    target="_blank"
  >
    <div />
    <img />
  </a>
</div>
`;
